core/renderer.ts

```diff
@@ -180,6 +180,14 @@ export default class Renderer {
       return;
     }
 
+    // Check that `url` is of a valid type
+    if (!["string", "function", "undefined"].includes(typeof url)) {
```

Comment:
This doesn't check the value returned when the url is a function, so a function that returns an integer would fail silently. I think the following flow works better:

```js
// Execute the function
if (typeof url === "function") {
  url = url(page);
}

if (typeof url === "string") {
  // The current code for strings
} else if (url !== undefined) {
  // throws an exception because the url is not valid
} else if (!this.formats.get(page.src.ext || "")?.asset) {
  // The current code for undefined url
}
```

Reply:
ah yes, got it! I was thinking of keeping the type validation closer to the edge, but had forgotten that the function's return type isn't enforced at runtime. Your suggested alternative sounds good to me 👍